CONTRIBUTING.md

```diff
@@ -36,11 +50,27 @@ for the Darwin and Linux platforms.
 ### Testing
 * To run unit tests, run `make test` from `$GOPATH/src/github.com/aws/amazon-ecs-cli`.
 
-### Licensing
-The Amazon ECS CLI is released under an [Apache 2.0](http://aws.amazon.com/apache-2-0/) license. Any code you submit will be released under that license.
+## Contributing code
+* Please check the existing issues and FAQ to see if your feedback has already been reported.
 
-For significant changes, we may ask you to sign a [Contributor License Agreement](http://en.wikipedia.org/wiki/Contributor_License_Agreement).
+* Let us know if you are interested in working on an issue by leaving a comment
+on the issue in GitHub. This helps avoid multiple people unknowingly working on
+the same issue.
+
+* If you would like to propose a new feature, please open an issue on GitHub with
+a detailed description. This enables us to collaborate on the feature design
+more easily and increases the chances that your feature request will be accepted.
+
+* New features should include full test coverage.
 
+* Ensure that your submissions have been formatted correctly by running `go fmt` on changed files.
+
+* All submissions, including submissions by project members, require review. We
+  use GitHub pull requests for this purpose. Consult GitHub Help for more
+information on using pull requests.
+
+* For any PR where you're updating `Gopkg.toml`, make sure to run `dep ensure
+  && dep prune` and commit all changes to vendor.
```

Comment:
Should we say "commit all changes as a separate commit"?

Reply:
All vendor update changes you mean?

Reply:
Yeah, updating vendor should probably be a separate commit within the same PR so that you can review the dependency update separately from the meaningful code change.